install/kubernetes/helm/istio/charts/mixer/templates/config.yaml

```diff
@@ -234,11 +240,24 @@ metadata:
 spec:
   value: "1"
   dimensions:
-    source_service: source.service | "unknown"
+    source_name: source.name | "unknown"
+    source_namespace: source.namespace | "unknown"
     source_workload: source.workload.name | "unknown"
+    source_workload_namespace: source.workload.namespace | "unknown"
+    source_principal: source.principal | "unknown"
+    source_app: source.labels["app"] | "unknown"
     source_version: source.labels["version"] | "unknown"
-    destination_service: destination.service | "unknown"
+    destination_name: destination.name | "unknown"
+    destination_namespace: destination.namespace | "unknown"
+    destination_workload: destination.workload.name | "unknown"
+    destination_workload_namespace: destination.workload.namespace | "unknown"
+    destination_principal: destination.principal | "unknown"
+    destination_app: destination.labels["app"] | "unknown"
     destination_version: destination.labels["version"] | "unknown"
+    destination_service: destination.service.host | "unknown"
+    destination_service_name: destination.service.name | "unknown"
+    destination_service_namespace: destination.service.namespace | "unknown"
+    request_protocol: api.protocol | context.protocol | "unknown"
```

Comment:
Looks like the cardinality of the default metrics will be way too high. Any idea on the default Prometheus's configuration capabilities for storage?

Reply:
why do you believe that the cardinality is impacted by this change?  large number of deployments?

Reply:
I guess I'm assuming that the # of unique combinations of all the source_* dimensions will be much higher than just the old # of unique source_services, and same for destinatinon_*. Maybe that is not the case. At least we will see increased cardinality based on pod name with the new set, as <src/dest>.name includes the workload instance name, correct?

I'd be nice if we kept the cardinality at: (# of workloads)^2 * (# of response codes)

Reply:
`source.service` does not exist any more. so our cardinality has gone up. however, i don't believe that the extra labels increase the cardinality beyond num_workloads too much (name, namespace, owner, and workload should not vary independently).  hopefully, `app` is more-or-less correlated with num_services and is 1:m(workloads).

nothing is currently using the `(source|destination).name` labels. I'd be fine with dropping that, as a way to reduce concerns here (as you could have a deployment that controls a *large* number of pods).

I'd also be fine with dropping `(source|destination).version` as that is highly-variable and somewhat obfuscated by the focus on workload. That just felt a bit radical.

Thoughts?